webapp/src/features/intersections/map/control-panel.tsx

```diff
@@ -67,9 +67,93 @@ import {
 import pauseIcon from '../../../icons/pause.png'
 import playIcon from '../../../icons/play.png'
 import { BarChart, XAxis, Bar, ResponsiveContainer, Tooltip } from 'recharts'
+import { useDispatch, useSelector } from 'react-redux'
+import { AnyAction, ThunkDispatch } from '@reduxjs/toolkit'
+import { RootState } from '../../../store'
 import { decoderModeToggled, setAsn1DecoderDialogOpen } from '../decoder/asn1-decoder-slice'
 import toast from 'react-hot-toast'
 
+const getNumber = (value: string | undefined): number | undefined => {
+  if (value == null) return undefined
+  const num = parseInt(value)
+  if (isNaN(num)) {
+    return undefined
+  }
+  return num
+}
+
+const formatMinutesAfterMidnightTime = (minutes) => {
```

Comment:
suggestion(non-blocking): define the type for minutes
```suggestion
const formatMinutesAfterMidnightTime = (minutes: number) => {
```